contents/docs/cdp/batch-exports/bigquery.md

```diff
@@ -0,0 +1,96 @@
+---
+title: BigQuery destination for batch exports
+sidebar: Docs
+showTitle: true
+availability:
+    free: full
+    selfServe: full
+    enterprise: full
+---
+
+Batch exports can be used to export data to a BigQuery table.
+
+## Setting up BigQuery access
+
+To set the right permissions up for a batch export targeting BigQuery, you will need:
+
+1. A Service Account.
+2. A dataset which has permissions allowing the service account to access it.
+
+Here's how to set these up so that the app has access only to the dataset it needs:
+
+1. Create a [Service Account](https://cloud.google.com/iam/docs/service-accounts-create#creating).
+
+![Create service account](../../images/docs/batch-exports/bigquery/create-service-account.png)
+
+2. Create a [key](https://cloud.google.com/iam/docs/keys-create-delete#creating) for the Service Account you created in the previous step.
+3. Save the key file as JSON to upload it when configuring a batch export.
+
+![Create JSON private key](../../images/docs/batch-exports/bigquery/create-private-key-json.png)
+
+4. Create a role which has only the specific permissions the batch export requires (listed below), or use the built in `BigQuery Data Owner` and `BigQuery Job User` roles. If you create a custom role, you will need:
+   * `bigquery.datasets.get`
+   * `bigquery.jobs.create`
+   * `bigquery.tables.create`
+   * `bigquery.tables.get`
+   * `bigquery.tables.list`
+   * `bigquery.tables.updateData`
+
+![Create custom role for batch exports](../../images/docs/batch-exports/bigquery/create-role.png)
+
+5. Grant the Service Account access to run jobs in your Google Cloud project. This can be done by granting the `BigQuery Jobs User` role on your project, or with the role we have just created in the previous step.
+
+Navigate to IAM and click on Grant Access to arrive at this screen:
+
+![Project grant](../../images/docs/batch-exports/bigquery/project-grant.png)
+
+In the screenshot above, we have used a custom role named `Testing PostHog BatchExports` with the permissions listed in the previous step.
+
+6. Create a dataset within a BigQuery project (ours is called `BatchExports`, but any name will do).
+7. Follow the instructions on [granting access to a dataset in BigQuery](https://cloud.google.com/bigquery/docs/control-access-to-resources-iam#grant_access_to_a_dataset):
+
+![Sharing dataset](../../images/docs/batch-exports/bigquery/dataset-sharing.png)
+![Add principal](../../images/docs/batch-exports/bigquery/dataset-add-principal.png)
+
+Use the Sharing and Add Principal buttons to share your dataset with your new Service Account created in step 1.
+
+Grant either the `BigQuery Data Owner` role, or your custom role created in step 4 to the Service Account. In the screenshot below, we have used a custom role named `Testing PostHog BatchExports`.
```

Comment:
Restructuring this to make it easier to follow. 
```suggestion
7. Use the Sharing and Add Principal buttons to share your dataset with your new Service Account created in step 1. Grant either the `BigQuery Data Owner` role or your custom role created in step 4 to the Service Account. Read the full instructions on [granting access to the dataset in BigQuery](https://cloud.google.com/bigquery/docs/control-access-to-resources-iam#grant_access_to_a_dataset) if unclear.

![Sharing dataset](../../images/docs/batch-exports/bigquery/dataset-sharing.png)
![Add principal](../../images/docs/batch-exports/bigquery/dataset-add-principal.png)

> In the screenshot below, we grant a custom role named `Testing PostHog BatchExports` access to the `BatchExports` data set.
```

Reply:
Not quite: The role only dictates the permissions (actions that the principal can do), but doesn't receive any grants. The Service Account (the principal) is the one getting assigned the permissions dictated in the `Testing PostHog BatchExports` role to operate on the BatchExports dataset. 

I may be making things more confusing, I definitely get lost with the right terminology.

Reply:
How about:

> In the screenshot below, we grant our Service Account access to the `BatchExports` data set and assign the `Testing PostHog BatchExports` role permissions for it.

Updated my suggestion with the change.

Reply:
Yeah, that sounds better 👍

Reply:
I can't commit the suggestion from here as it's marked as out of date, but I'll make a new commit locally with the same changes.

Reply:
Done! Thank you!